src/extension/mapml/src/test/java/org/geoserver/mapml/MapMLWMTSProxyTest.java

```diff
@@ -0,0 +1,160 @@
+        namespaces.put("xlink=", "http://www.w3.org/1999/xlink");
+    }
+
+    @BeforeClass
+    public static void beforeClass() {
+        initMockService(
+                "/mockgeoserver",
+                "/gwc/service/wmts",
+                "REQUEST=GetCapabilities&VERSION=1.0.0&SERVICE=WMTS",
+                "wmtscaps.xml");
+    }
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        Catalog catalog = getCatalog();
+
+        WMTSStoreInfo wmtsStore = catalog.getFactory().createWebMapTileServer();
+        wmtsStore.setName("wmtsStore");
+        wmtsStore.setWorkspace(catalog.getDefaultWorkspace());
+        wmtsStore.setCapabilitiesURL(
+                "http://localhost:" + mockService.port() + getCapabilitiesURL());
+        wmtsStore.setEnabled(true);
+        catalog.add(wmtsStore);
+
+        // Create WMSLayerInfo using the Catalog factory
+        WMTSLayerInfo wmtsLayer = catalog.getFactory().createWMTSLayer();
+        wmtsLayer.setName("cascadedLayer");
+        wmtsLayer.setNativeName("topp:states");
+        wmtsLayer.setStore(wmtsStore);
+        wmtsLayer.setAdvertised(true);
+        wmtsLayer.setEnabled(true);
+
+        // Add the layer to the catalog
+        LayerInfo layerInfo = catalog.getFactory().createLayer();
+        layerInfo.setResource(wmtsLayer);
+        layerInfo.setDefaultStyle(catalog.getStyleByName("default"));
+        catalog.add(wmtsLayer);
+        catalog.add(layerInfo);
+
+        GWC gwc = applicationContext.getBean(GWC.class);
+        GWCConfig defaults = GWCConfig.getOldDefaults();
+        // it seems just the fact of retrieving the bean causes the
+        // GridSets to be added to the gwc GridSetBroker, but if you don't do
+        // this, they are not added automatically
+        MapMLGridsets mgs = applicationContext.getBean(MapMLGridsets.class);
+        GridSubset wgs84gridset = createGridSubSet(mgs.getGridSet("WGS84").get());
+        GridSubset osmtilegridset = createGridSubSet(mgs.getGridSet("OSMTILE").get());
+
+        GeoServerTileLayer layerInfoTileLayer =
+                new GeoServerTileLayer(layerInfo, defaults, gwc.getGridSetBroker());
+        layerInfoTileLayer.addGridSubset(wgs84gridset);
+        layerInfoTileLayer.addGridSubset(osmtilegridset);
+        layerInfoTileLayer.getInfo().getMimeFormats().add(TextMime.txtMapml.getMimeType());
+        gwc.save(layerInfoTileLayer);
+    }
+
+    @Test
+    public void testRemoteVsNotRemote() throws Exception {
+        Catalog cat = getCatalog();
+        // Verify the layer was added
+        LayerInfo li = cat.getLayerByName("cascadedLayer");
+        assertNotNull(li);
+        assertEquals("cascadedLayer", li.getName());
+
+        ResourceInfo layerMeta = li.getResource();
+        layerMeta.getMetadata().put("mapml.useRemote", false);
+        layerMeta.getMetadata().put("mapml.useTiles", true);
```

Comment:
This one should have a constant already available too I believe (not verified though).